### DISCUSSION TITLE
Checkbox Arrays Not Working

### DISCUSSION THREAD

**Discussion by sgtcoder** [spatie/laravel-html#191] [Q&A]
There is an issue when loading model data that contains a checkbox array in a form. I know I can do the `in_array` function, but that would be tedious for a couple hundred checkboxes. I looked at the source code and it seems like it only checks the value, rather than also handling multiple/array.

Here is an example code:

```
{{ html()->model($order)->form('POST')->open() }}

{{ html()->checkbox('sandwiches[]', null, 'ham') }}

{{ html()->checkbox('sandwiches[]', null, 'turkey') }}

{{ html()->form()->close() }}
```

**Comment by andjelko** [comment]
You can use method "checked" like:

`{{ html()->checkbox('sandwiches[]', null, 'turkey')->checked(isset($order) ? $order->hasSandwiches( 'turkey') : 0 ) }}
`

And than in model "Order" you can write function "hasSandwiches" like this:
```
public function hasSandwiches($sandwich_name)
{
    foreach ($this->sandwiches as $sandwich){
        if ($sandwich->name == $sandwich_name)
            return true;
    }
    return false;
}
```

**Reply by sgtcoder** [reply]
Yes, I know I can use checked. But, Laravel Collectives had this built-in.
